lib/ExpensiMark.js

```diff
@@ -137,4 +149,46 @@ export default class ExpensiMark {
 
         return replacedText;
     }
+
+    /**
+     * Checks matched URLs for validity and replace valid links with html elements
+     *
+     * @param {RegExp} regex
+     * @param {String} textToCheck
+     * @param {Function} replacement
+     *
+     * @returns {String}
+     */
+    modifyTextForUrlLinks(regex, textToCheck, replacement) {
```

Comment:
This function is specifically written for our URL regex, so IMO we should not be passing it here.

Reply:
The regex is not the same since this is used for both the `link` and `autolink` rules. That's why it's being passed to the function.

Reply:
Ah yes, forgot those two were using differennt regexes. This is good then!